bumble/device.py

```diff
@@ -5449,15 +5411,58 @@ def _complete_le_extended_advertising_connection(
         self.emit(self.EVENT_CONNECTION, connection)
 
     @host_event_handler
-    def on_connection(
+    def on_connection_complete(
+        self,
+        connection_handle: int,
+        peer_address: hci.Address,
+        connection_interval: int,
+        peripheral_latency: int,
+        supervision_timeout: int,
+    ) -> None:
+        connection_role = self.connection_roles.pop(peer_address)
```

Comment:
Still safer to have a fallback
```suggestion
        connection_role = self.connection_roles.pop(peer_address, hci.Role.PERIPHERAL)
```